native/ax/MpegDeMux-3/CDemuxer.cpp

```diff
@@ -953,7 +956,7 @@ HRESULT CDemuxer::SetMpeg2VideoMediaType(CMediaType *cmt, MPEG_VIDEO *pMpegVideo
 	
 	videoInfo->hdr.dwInterlaceFlags = pMpegVideo->progressive ? 0 : (AMINTERLACE_IsInterlaced);
 	videoInfo->hdr.AvgTimePerFrame = pMpegVideo->picture_time;
-    videoInfo->cbSequenceHeader = pMpegVideo->actual_header_length;
+    videoInfo->cbSequenceHeader = header_padded_length;
```

Comment:
I agree with allocating more...but should you be setting the size to the aligned size and not the actual size?  Did you test if that works OK?

Reply:
I thought you would notice that.  ;)  I suppose I should have included that in my first comment.

All of my testing showed that the cbSequenceHeader was really what needed to be DWORD aligned.  It still crashed even when I allocated and zeroed lots of extra buffer size but left cbSequenceHeader alone.  There must be something about accessing or freeing the memory that makes use of cbSequenceHeader.  Since the header information is formatted and parsed I imagine it wouldn't matter if there are pad bytes after it.

The commit I submitted was the one I tested.

Reply:
OK, that's good enough for me. :)  If you wouldn't mind making those statements cleaner with an ALIGN macro, that'd be great and then I'll approve it for merge.

Reply:
Done.  I named it ALIGN_DWORD since it's specific to a size of 4.